packages/renderer/src/ffmpeg-flags.ts

```diff
@@ -51,16 +126,178 @@ export const parseFfmpegVersion = (buildconf: string): FfmpegVersion => {
 
 export const getFfmpegVersion = async (options: {
 	ffmpegExecutable: string | null;
+	remotionRoot: string;
 }): Promise<FfmpegVersion> => {
-	const buildInfo = await getFfmpegBuildInfo({
-		ffmpegExecutable: options.ffmpegExecutable,
-	});
+	const buildInfo = await getFfmpegBuildInfo(
+		{
+			ffmpegExecutable: options.ffmpegExecutable,
+		},
+		options.remotionRoot
+	);
 	return parseFfmpegVersion(buildInfo);
 };
 
+const waitForFfmpegToBeDownloaded = (url: string) => {
+	return new Promise<string>((resolve) => {
+		if (!listeners[url]) {
+			listeners[url] = [];
+		}
+
+		listeners[url].push((src) => resolve(src));
+	});
+};
+
+const onProgress = (downloadedBytes: number, totalBytesToDownload: number) => {
+	console.log(
+		'Downloading FFmpeg: ',
+		`${toMegabytes(downloadedBytes)}/${toMegabytes(totalBytesToDownload)}`
+	);
+};
+
+export const downloadBinary = async (
+	remotionRoot: string,
+	url: string,
+	binary: 'ffmpeg' | 'ffprobe'
+): Promise<string> => {
+	const destinationPath = getFfmpegAbsolutePath(remotionRoot, binary);
+
+	isDownloading[url] = true;
+	const totalBytes = await _downloadFile(url, destinationPath, onProgress);
+	onProgress(totalBytes, totalBytes);
+	if (os.platform() !== 'win32') {
+		fs.chmodSync(destinationPath, '777');
+	}
+
+	isDownloading[url] = false;
+	if (!listeners[url]) {
+		listeners[url] = [];
+	}
+
+	listeners[url].forEach((listener) => listener(destinationPath));
+	listeners[url] = [];
+
+	return destinationPath;
+};
+
+export const getExecutableFfprobe = async (
+	ffprobeExecutable: FfmpegExecutable,
+	remotionRoot: string
+) => {
+	const exists = binaryExists('ffprobe', ffprobeExecutable);
+
+	if (exists) {
+		if (ffprobeExecutable !== null) {
+			return ffprobeExecutable;
+		}
+
+		return 'ffprobe';
+	}
+
+	const {url} = getFfmpegDownloadUrl('ffprobe');
+
+	if (isDownloading[url]) {
+		return waitForFfmpegToBeDownloaded(url);
+	}
+
+	const inNodeMod = ffmpegInNodeModules(remotionRoot, 'ffprobe');
+
+	if (inNodeMod) {
+		return inNodeMod;
+	}
+
+	return downloadBinary(remotionRoot, url, 'ffprobe');
+};
+
+export const getExecutableFfmpeg = async (
```

Comment:
This and getExecutableFfprobe has the exact same logic, so i'd propose to have only 1 function